test/exporter/test_cqasmv1_exporter.py

```diff
@@ -236,3 +236,57 @@ def test_anonymous_gates(gate: Gate) -> None:
     with pytest.raises(UnsupportedGateError, match="not supported"):  # noqa: PT012
         qc = builder.to_circuit()
         qc.export(fmt=ExportFormat.CQASM_V1)
+
+
+@pytest.mark.parametrize(
+    ("v3_input", "v3_output", "v1_output"),
+    [
+        (
+            """version 3.0; qubit[1] q; barrier q[0]; """,
+            """version 3.0\n\nqubit[1] q\n\nbarrier q[0]\n""",
+            """version 1.0\n\nqubits 1\n\nbarrier q[0]\n""",
+        ),
+        (
+            """version 3.0; qubit[3] q; barrier q[0:2]; """,
+            """version 3.0\n\nqubit[3] q\n\nbarrier q[0]\nbarrier q[1]\nbarrier q[2]\n""",
+            """version 1.0\n\nqubits 3\n\nbarrier q[0, 1, 2]\n""",
+        ),
+        (
+            """version 3.0; qubit[1] q; barrier q[0, 0];""",
+            """version 3.0\n\nqubit[1] q\n\nbarrier q[0]\nbarrier q[0]\n""",
+            """version 1.0\n\nqubits 1\n\nbarrier q[0, 0]\n""",
+        ),
+        (
+            """version 3.0; qubit[6] q; barrier q[0:2, 5, 3, 4, 1]; """,
+            """version 3.0\n\nqubit[6] q\n\nbarrier q[0]\nbarrier q[1]\nbarrier q[2]\nbarrier q[5]\nbarrier q[3]
+barrier q[4]\nbarrier q[1]\n""",
+            """version 1.0\n\nqubits 6\n\nbarrier q[0, 1, 2, 5, 3, 4, 1]\n""",
+        ),
+        (
+            """version 3.0; qubit[5] q; barrier q[0]; H q[1]; barrier q[1:2]; X q[2]; barrier q[3:4, 1]; Y q[3];
+            barrier q[0];""",
+            """version 3.0\n\nqubit[5] q\n\nbarrier q[0]\nH q[1]\nbarrier q[1]\nbarrier q[2]\nX q[2]
+barrier q[3]\nbarrier q[4]\nbarrier q[1]\nY q[3]\nbarrier q[0]\n""",
+            """version 1.0\n\nqubits 5\n\nbarrier q[0]\nh q[1]\nbarrier q[1, 2]\nx q[2]\nbarrier q[3, 4, 1]\ny q[3]
+barrier q[0]\n""",
+        ),
+        (
+            """version 3.0; qubit[3] q; barrier q[0]; barrier q[1]; X q[2]; barrier q[1]""",
+            """version 3.0\n\nqubit[3] q\n\nbarrier q[0]\nbarrier q[1]\nX q[2]\nbarrier q[1]\n""",
+            """version 1.0\n\nqubits 3\n\nbarrier q[0, 1]\nx q[2]\nbarrier q[1]\n""",
+        ),
```

Comment:
We shouldn't need:
- triple quotes for single line texts.
- semicolons, whitespaces, or newlines at the end of an input cQASM program.

Double check if exported programs need the newline at the end.

```suggestion
            """version 3.0; qubit[1] q; barrier q[0]",
            """version 3.0\n\nqubit[1] q\n\nbarrier q[0]\n",
            """version 1.0\n\nqubits 1\n\nbarrier q[0]\n",
        ),
        (
            "version 3.0; qubit[3] q; barrier q[0:2]",
            "version 3.0\n\nqubit[3] q\n\nbarrier q[0]\nbarrier q[1]\nbarrier q[2]\n",
            "version 1.0\n\nqubits 3\n\nbarrier q[0, 1, 2]\n",
        ),
        (
            "version 3.0; qubit[1] q; barrier q[0, 0]",
            "version 3.0\n\nqubit[1] q\n\nbarrier q[0]\nbarrier q[0]\n",
            "version 1.0\n\nqubits 1\n\nbarrier q[0, 0]\n",
        ),
        (
            "version 3.0; qubit[6] q; barrier q[0:2, 5, 3, 4, 1]",
            "version 3.0\n\nqubit[6] q\n\nbarrier q[0]\nbarrier q[1]\nbarrier q[2]\nbarrier q[5]\nbarrier q[3]
barrier q[4]\nbarrier q[1]\n",
            "version 1.0\n\nqubits 6\n\nbarrier q[0, 1, 2, 5, 3, 4, 1]\n",
        ),
        (
            "version 3.0; qubit[5] q; barrier q[0]; H q[1]; barrier q[1:2]; X q[2]; barrier q[3:4, 1]; Y q[3];
            barrier q[0]",
            "version 3.0\n\nqubit[5] q\n\nbarrier q[0]\nH q[1]\nbarrier q[1]\nbarrier q[2]\nX q[2]
barrier q[3]\nbarrier q[4]\nbarrier q[1]\nY q[3]\nbarrier q[0]\n",
            "version 1.0\n\nqubits 5\n\nbarrier q[0]\nh q[1]\nbarrier q[1, 2]\nx q[2]\nbarrier q[3, 4, 1]\ny q[3]
barrier q[0]\n",
        ),
        (
            "version 3.0; qubit[3] q; barrier q[0]; barrier q[1]; X q[2]; barrier q[1]",
            "version 3.0\n\nqubit[3] q\n\nbarrier q[0]\nbarrier q[1]\nX q[2]\nbarrier q[1]\n",
            "version 1.0\n\nqubits 3\n\nbarrier q[0, 1]\nx q[2]\nbarrier q[1]\n",
        ),
```

Reply:
For the future, when indices and SGMQ are taken care of, a couple of cases should be looked into:
1. `q[0, 0]`. Wouldn't that be the same as `q[0]`?
2. `q[3, 2, 1]`. Better written as `q[1, 2, 3]`. We cannot do that at the moment for things like `CNOT q[1, 2] q[4, 0]`, where order is important.

Reply:
I'll add this to your document on SGMQ notation, so that we can take it into account when we deal with that.